financial-templates-lib/clients/ExpiringMultiPartyEventClient.js

```diff
@@ -113,17 +113,50 @@ class ExpiringMultiPartyEventClient {
   async update() {
     // The last block to search is either the value specified in the constructor (useful in serverless mode) or is the
     // latest block number (if running in loop mode).
-    const lastBlockToSearch = this.lastBlockToSearchUntil
-      ? this.lastBlockToSearchUntil
-      : await this.web3.eth.getBlockNumber();
-
-    // Look for events on chain from the previous seen block number to the current block number.
-    // Liquidation events
-    const liquidationEventsObj = await this.emp.getPastEvents("LiquidationCreated", {
+    const [currentBlockNumber, currentTime] = await Promise.all([
+      this.web3.eth.getBlockNumber(),
+      this.emp.methods.getCurrentTime().call()
```

Comment:
I think we should move the `getCurrentTime()` promise with the last Promise.all in this function so it moreaccurately reflects the last update time.

Reply:
ye, good point. I have implemented the change.

The monitor bot also implements this client and uses it to look for liquidations, disputes, dispute settlements & new sponsor events.

I have not validated that the daily reporter still runs but all the unit tests still work fine. I can run it to double-check but feel pretty confident seeing the tests all pass. What do you think?

Reply:
LGTM but yeah if possible can you run reporter locally as sanity check?

Reply:
Tested and still workes exactly as expected :)